iris-mpc-gpu/src/server/actor.rs

```diff
@@ -717,7 +719,7 @@ impl ServerActor {
         ///////////////////////////////////////////////////////////////////
         // PERFORM DELETIONS (IF ANY)
         ///////////////////////////////////////////////////////////////////
-        if !batch.deletion_requests_indices.is_empty() {
+        if !batch.deletion_requests_indices.is_empty() && orientation == Orientation::Normal {
```

Comment:
This is a problem, since if we do not process the deletion requests in the mirror flow already, the two flows have different views of the DB. (Mirror flow has unmodified DB, Normal ones has the DB with deletions). I do not know if that can present an issue in practice though?